actions/cloud.tsx

```diff
@@ -0,0 +1,98 @@
+
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import {Stripe} from '@stripe/stripe-js';
+import {getCode} from 'country-list';
+
+import {Client4} from 'mattermost-redux/client';
+import {Product} from 'mattermost-redux/types/cloud';
+
+import {getConfirmCardSetup} from 'components/payment_form/stripe';
+
+import {StripeSetupIntent, BillingDetails} from 'components/cloud/types/sku';
+
+export function getProductPrice() {
+    return async () => {
+        let cloudProducts = [] as Product[];
+
+        try {
+            cloudProducts = await Client4.getCloudProducts();
+        } catch (error) {
+            // eslint-disable-next-line no-console
+            console.error(`Error fetching cloud products: ${error}`);
+        }
+
+        let productPrice = 0;
+        if (cloudProducts.length > 0) {
+            // Assuming the first and only one for now.
+            productPrice = cloudProducts[0].price_per_seat;
+        }
+
+        return productPrice;
+    };
+}
+
+export function completeStripeAddPaymentMethod(stripe: Stripe, billingDetails: BillingDetails, isDevMode: boolean) {
+    return async () => {
+        let paymentSetupIntent: StripeSetupIntent;
+        try {
+            paymentSetupIntent = await Client4.createPaymentMethod() as StripeSetupIntent;
+        } catch (error) {
+            console.error(error); //eslint-disable-line no-console
+            return error;
+        }
+        const cardSetupFunction = getConfirmCardSetup(isDevMode);
+        const confirmCardSetup = cardSetupFunction(stripe.confirmCardSetup);
+
+        const result = await confirmCardSetup(
+            paymentSetupIntent.client_secret,
+            {
+                payment_method: {
+                    card: billingDetails.card,
+                    billing_details: {
+                        name: billingDetails.name,
+                        address: {
+                            line1: billingDetails.address,
+                            line2: billingDetails.address2,
+                            city: billingDetails.city,
+                            state: billingDetails.state,
+                            country: getCode(billingDetails.country),
+                            postal_code: billingDetails.postalCode,
+                        },
+                    },
+                },
+            },
+        );
+
+        if (!result) {
+            console.error('Stripe confirm card setup failed.'); //eslint-disable-line no-console
+            return 'Stripe confirm card failed';//'intl.formatMessage({id: 'errors.generic_payment_failure'})'};
```

Comment:
We still need to add localization for these error messages in this function

Reply:
I thought we were saving these for the console log, and showing a generic error to the user? I don't think it's super helpful to tell the end user that we're using stripe. Should I make them more generic? Or just show the same generic error?

Reply:
The console error logs is fine but if we are returning an error here and displaying it somewhere then we need to localize that. If not, then does it make sense to return an error at all? We can return a generic error like the one commented next to it.

Reply:
I've changed this function up. It logs the error if it runs into one, but displays generic error to the user. This now returns boolean true/false for success/failure respectively. Let me know what you think now!